src/user/Admin/Tickets/TicketDiscussion/Layout/Layout.js

```diff
@@ -0,0 +1,188 @@
+import React, { Component } from "react";
+import Tags from "./Tags";
+import Title from "./Title";
+import TicketInfo from "./Info";
+import Members from "./Members";
+import Card from "react-bootstrap/Card";
+import Image from "react-bootstrap/Image";
+import EditableCard from "./EditableCard";
+import Tooltip from "@material-ui/core/Tooltip";
+import LoadingOverlay from "react-loading-overlay";
+import ClockLoader from "react-spinners/ClockLoader";
+import { withStyles } from "@material-ui/core/styles";
+import userIcon2 from "../../../../../assets/images/userIcon2.jpg";
+import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined";
+
+class Layout extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tagsEditor: false,
+      titleEditor: false,
+      summaryEditor: false,
+      tags: props.ticket?.tags,
```

Comment:
Please not use props in state

Reply:
Why? The state in this component needs to have an initial value that is passed down dynamically from parent components. This seems to be a standard thing to do, is this a bad practice? Can you suggest any work around? Or share some resources? Thanks

Reply:
Have a look at it: https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e

So you should use lifecycle methods to initialize if you want to initialize via props.

Reply:
ComponentDidMount

Reply:
Thanks, Done